### PR TITLE
Feature/pin comment

### PR DESCRIPTION
**Pull request by teamchz**
- Creating pin comments that are always on the top of the comment section using bloc inside comment bloc files
- Make pin real-time by adding it into the bloc whether the response is 200 or else
- Fix Sort to can work with pinned comment
- Rename some variables to make them cleaner

### CODE REVIEW

**Review by thunyathep — COMMENTED** [review]
GJ โบ๋

**Review by Balloii — APPROVED** [review]

**Review by thunyathep — APPROVED** [review]

File: lib/src/blocs/comment/comment_bloc.dart [hunk 1]
```diff
@@ -78,22 +78,56 @@ class CommentBloc extends Bloc<CommentEvent, CommentState> {
...
-      listCommentModel.comments = listCommentModel.comments.reversed.toList();
+      listCommentModel.comments.sort((comment1, comment2) {
+        if (comment2.isPin == comment1.isPin) {
+          return convertDate(comment2.Date)
+              .compareTo(convertDate(comment1.Date));
+        } else {
+          return comment2.isPin! ? 1 : -1; // true values come first
+        }
+      });
+
```
Comment (aofattaporn) [inline lib/src/blocs/comment/comment_bloc.dart:89]: why do you need to return 1 or -1, I think in actual return a boolean is the best way

**Review by aofattaporn — COMMENTED** [review]
I have some questions, pls, answer me.

File: lib/src/blocs/comment/comment_bloc.dart (hunk 1 above)
Comment (teamchz) [reply lib/src/blocs/comment/comment_bloc.dart:89]: This function does not set "comment.isPin" to True or False but indicates the order of the two comments being compared (sort by comparing which one should come before). And importantly, they required integer values, so if use a boolean instead, the function would no longer work correctly.

Comment (aofattaporn) [reply lib/src/blocs/comment/comment_bloc.dart:89]: ok, It sounds reasonable!

**Comment by aofattaporn** [comment]
LGTM

**Pull request merged**